src/Integration.Vsix/SonarLintTracker/IssuesSnapshot.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace SonarLintTracker
+{
+    class IssuesSnapshot : WpfTableEntriesSnapshotBase
+    {
+        private readonly string filePath;
+        private readonly int versionNumber;
+
+        // We're not using an immutable list here but we cannot modify the list in any way once we've published the snapshot.
+        public readonly List<IssueMarker> IssueMarkers = new List<IssueMarker>();
+
+        public IssuesSnapshot NextSnapshot;
+
+        internal IssuesSnapshot(string filePath, int versionNumber)
+        {
+            this.filePath = filePath;
+            this.versionNumber = versionNumber;
+        }
+
+        public override int Count
+        {
+            get
+            {
+                return this.IssueMarkers.Count;
+            }
+        }
+
+        public override int VersionNumber
+        {
+            get
+            {
+                return versionNumber;
+            }
+        }
+
+        public override bool TryGetValue(int index, string columnName, out object content)
+        {
+            if ((index >= 0) && (index < this.IssueMarkers.Count))
+            {
+                // TODO missing issue type, such as bug, vulnerability, code smell
+                if (columnName == StandardTableKeyNames.DocumentName)
+                {
+                    content = filePath;
+                    return true;
+                }
+                else if (columnName == StandardTableKeyNames.Line)
+                {
+                    content = this.IssueMarkers[index].Span.Start.GetContainingLine().LineNumber;
+                    return true;
+                }
+                else if (columnName == StandardTableKeyNames.Column)
+                {
+                    var position = this.IssueMarkers[index].Span.Start;
+                    var line = position.GetContainingLine();
+                    content = position.Position - line.Start.Position;
+                    return true;
+                }
+                else if (columnName == StandardTableKeyNames.Text)
+                {
+                    content = this.IssueMarkers[index].Issue.Message;
+                    return true;
+                }
+                else if (columnName == StandardTableKeyNames.ErrorSeverity)
+                {
+                    content = __VSERRORCATEGORY.EC_WARNING;
+                    return true;
+                }
+                else if (columnName == StandardTableKeyNames.BuildTool)
+                {
+                    // TODO get correct analyzer name
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)